apps/hash-graph/bin/cli/src/subcommand/snapshot.rs

```diff
@@ -49,36 +48,66 @@ pub async fn snapshot(args: SnapshotArgs) -> Result<(), GraphError> {
             report
         })?;
 
-    let mut store = pool
-        .acquire()
-        .await
-        .change_context(GraphError)
-        .map_err(|report| {
-            tracing::error!(error = ?report, "Failed to acquire database connection");
-            report
-        })?;
+    let mut store = SnapshotStore::new(
+        pool.acquire_owned()
+            .await
+            .change_context(GraphError)
+            .map_err(|report| {
+                tracing::error!(error = ?report, "Failed to acquire database connection");
+                report
+            })?,
+    );
+
+    // TODO: Use a thin wrapper around serde-json which implements `Sink` and `Stream` to avoid the
+    //       need for a separate thread.
+    let (sender, mut receiver) = futures::channel::mpsc::channel::<SnapshotEntry>(8);
```

Comment:
It's randomly chosen. Maybe it should be `4`? 😂 
Any number should work (but I didn't want to use `0` or an unbounded channel to have some backpressure), so I chose 8. As I knew I was about to remove it in a follow-up I didn't think much about it.

Reply:
If we're removing it then no worries, I was guessing we'd want quite a much more than `8` actually 😂

Reply:
To explain this a little bit more: this is just the buffer the channel can hold until it's blocking any further item to be pushed into the channel. E.g. for dumping the snapshot it can hold 8 entries before they has to be written to `stdout`. Having a low number implies that everything needs to be printed to stdout immediately (or at least written the buffer held by `BufWriter`). A high number or unbounded channels could imply that the database creates a lot of entries, which have not been written to `stdout` yet, so they have to be kept in memory.

Reply:
I went and read the `docs.rs` page and somehow managed to convince myself the buffer was in bytes, as opposed to num entries.. Thanks for clarifying